wrappers/desktop.go

```diff
@@ -260,6 +260,9 @@ func deriveDesktopFilesContent(s *snap.Info) (map[string]osutil.FileState, error
 		if err != nil {
 			return nil, err
 		}
+		if _, exists := content[base]; exists {
```

Comment:
IIRC this is called indirectly from Ensure(), what if a snap where this problem exists is already installed in the system?

Reply:
It is okay because on startup snapd re-writes desktop files for all snaps.
https://github.com/canonical/snapd/blob/8774c075d963078a649795b61370aa3404971bc8/overlord/snapstate/snapmgr.go#L1600

Reply:
Good point, it would fail. Do you think it would be better to log and ignore duplicates?

Reply:
preferrably yes

Reply:
updated, thanks!